src/images/maps/map_test.R

```diff
@@ -48,16 +53,62 @@ map_test <- function(
       legend.position = "left",
       legend.direction = "vertical"
     )
+  return(p)
+  #
+  #   iso3_ggsave$iso3_ggsave(
+  #     p = p,
+  #     iso3 = iso3,
+  #     fp = file.path(
+  #       dir,
+  #       paste0(iso3, ".png")
+  #     ),
+  #     crop = FALSE
+  #   )
+  #
+  #   invisible(NULL)
+}
+
+
+#' Generate pts for test map
+#'
+#' @param poly
+#' @param number_pt_range `integer` vector (default = 1:20) sampled to randomly decide how many
+#'  pts will be mapped
+#' @param value_range `integer` vector (defaults 1:20000) sampled to assign values to pts.
+#'  might be useful to have it sample a log distribution instead
+#' @param use_bbox `logical` if TRUE (default) samples will be drawn from the bounding box
+#'  of the polygon, if FALSE sampled withing the polygon. It's mainly a feature to
+#'  ease development as sampling withing bbox is much faster
+#' @return
+#' @export
+#'
+#' @examples \dontrun{
+#' box::use(sf)
+#' box::use(../../utils/get_iso3_sf)
+#' gdf_adm0 <- get_iso3_sf$get_iso3_sf("AFG")
+#' pts_sampled_bbox <- gen_pts(poly = gdf_adm0,
+#'                             use_bbox=T,
+#'                              number_pt_range= 1:20,
+#'                              value_range= 1:20000)
+#' }
+
+random_spatial_sample <- function(poly,
+                                  use_bbox =T,
+                                  number_pt_range= 1:20,
+                                  value_range= 1:20000
+){
+  num_pts <- sample(x= number_pt_range,size = 1,replace=T)
+
+  if(use_bbox){
+    gdf_region <- sf$st_bbox(poly) |>
+      sf$st_as_sfc()
+  } else{
+    gdf_region <- poly
+  }
+
+  pts_gdf <- sf$st_sample(x=gdf_region,size =num_pts) |>
+    sf$st_as_sf()
 
-  iso3_ggsave$iso3_ggsave(
-    p = p,
-    iso3 = iso3,
-    fp = file.path(
-      dir,
-      paste0(iso3, ".png")
-    ),
-    crop = FALSE
-  )
-
-  invisible(NULL)
+  pts_gdf$value <- sample(value_range, num_pts, replace = T)
+  return(pts_gdf)
```

Comment:
Only use `return()` if exiting the function early, based on the Tidyverse style guide, which I quite like and find clean, and allows for clear and easy piping.

Reply:
oh interesting - seems nice, i've never heard that convention

Reply:
[8e43ed](https://github.com/OCHA-DAP/hdx-signals/pull/75/commits/88e43ed3e1e282b441d8c00d9937e6b55a7b9ad7)